CI: Add linkcheckers

TomVasile opened the pull request.

ipopescu reviewed (approved):
1. I ran the site locally and it renders correctly.
2. I combed through the list of broken links and fixed them. 
3. I also updated links that point to old repositories.

TomVasile commented:
> 1. I ran the site locally and it renders correctly.
> 2. I combed through the list of broken links and fixed them.
> 3. I also updated links that point to old repositories.

@ipopescu 
CI is saying these are still broken
```
(archived/contract-dsl-archived/getting-started: line   27) broken    https://github.com/CasperLabs/casperlabs_contract_macro - 404 Client Error: Not Found for url: https://github.com/CasperLabs/casperlabs_contract_macro
(dapp-dev-guide/tutorials/multi-sig/client: line   10) broken    https://github.com/network/casper-node/tree/master/utils/nctl - 404 Client Error: Not Found for url: https://github.com/network/casper-node/tree/master/utils/nctl
```

ipopescu commented:
@ramujin, I have added you to let you know that I've made these updates and fixed old/broken links. Thanks!
